### ISSUE TITLE
Digest cache should be per-request in development mode

### ISSUE THREAD

**Issue by dhh** [rails/rails#20361]
Currently we compute the template digest on every template lookup in development mode. This allows you to change the template and see the change right away. That's great. 

But in cases where you're looping over a bunch of the same templates within a single request, we compute the same digests over and over again within that request. That's not necessary, and it really slows down things in local development (and spams the shit out of the log).

So, let's instead have a per-request cache for the template digests. Still allows you to make changes between requests and instantly see the changes, but doesn't compute the same digests a million times needlessly within the request either.

**Comment by dhh** [comment]
@kaspth Is this something you might want to take a look at? Given that you're sorta in that neck of the wood with the cache multi loader.

**Comment by kaspth** [comment]
I'd rephrase that as "deep in the nightmare forest chased by an axe murderer" :trollface:

But yeah, I'll take a look because this has bothered me too :+1:

--
Kasper

**Comment by dhh** [comment]
Hahaha, that was a 10K XP mission right there.

**Comment by rafaelfranca** [comment]
lol. Since we are here, I was talking a look on this code with @britto and he has some inputs about it.
